core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -367,10 +367,14 @@ void print_stack(int level)
 {
 	struct unwind_state state;
 
-	memset(&state, 0, sizeof(state));
+	memset(state.registers, 0, sizeof(state.registers));
+	/* r7: Thumb-style frame pointer */
+	state.registers[7] = read_r7();
```

Comment:
I'd rather not for two reasons: (1) it does not improve legibility and (2) the above code is mostly copied from FreeBSD so the less we change it, the better.